tests/kernel/timer/timer_monotonic/testcase.yaml

```diff
@@ -4,6 +4,12 @@ tests:
     # FIXME: This test may fail for qemu_arc_hs on certain host systems.
     #        See foss-for-synopsys-dwc-arc-processors/qemu#67.
     platform_exclude: qemu_arc_hs
+  kernel.timer.monotonic.apic.tsc:
+     tags: kernel timer apic_tsc
+     platform_allow: ehl_crb
```

Comment:
Maybe add up_squared et. al. here?  Basically all x86 hardware (but not qemu, frustratingly) supports this driver.

Alternatively: just change the platform to use TSC_DEADLINE by default and deprecate HPET on EHL?  What's holding things back?

Reply:
> Maybe add up_squared et. al. here? Basically all x86 hardware (but not qemu, frustratingly) supports this driver.
> 
> Alternatively: just change the platform to use TSC_DEADLINE by default and deprecate HPET on EHL? What's holding things back?

The testing can default to do testing the HPET timer by using platform configuration. Here would like to testing TSC_DEADLINE timer even though the default timer is HPET timer. 

Yes, we also can enable TSC_DEADLINE on platform configuration to run the testing manually.  But I would like to cover the TSC_DEADLINE testing automatically.